client/src/components/JurisdictionAdmin/BatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,645 @@
+                        areResultsFinalized={areResultsFinalized}
+                        cancelSheetRename={cancelSheetRename}
+                        confirmSheetRename={confirmSheetRename}
+                        contest={contest}
+                        deleteSheet={() => deleteSheet(i)}
+                        enableEditing={enableEditing}
+                        formMethods={formMethods}
+                        index={i}
+                        initiateSheetRename={() => initiateSheetRename(i)}
+                        isEditing={isEditing}
+                        isRenamingSheet={isRenamingSheet}
+                        savedResults={
+                          selectedBatch.resultTallySheets[i]?.results || {}
+                        }
+                        saveResults={saveResults}
+                        sheets={sheets}
+                      />
+                    }
+                  >
+                    {sheetNameBeforeRename && sheetNameBeforeRename.index === i
+                      ? sheetNameBeforeRename.name
+                      : watch(`resultTallySheets[${i}].name`)}
+                  </Tab>
+                ))}
+
+              <Tabs.Expander />
+
+              {isEditing && (
+                <Button
+                  disabled={isRenamingSheet}
+                  icon="add"
+                  minimal
+                  onClick={addSheet}
+                >
+                  {sheets.length > 1
+                    ? 'Add sheet'
+                    : 'Use multiple tally sheets'}
+                </Button>
+              )}
+            </Tabs>
+          </>
+        )}
+      </Detail>
+      <Confirm {...confirmProps} />
+    </ListAndDetail>
+  )
+}
+
+interface IBatchResultTallySheetProps {
+  areResultsFinalized: boolean
+  cancelSheetRename: () => void
+  confirmSheetRename: () => void
+  contest: IContest
+  deleteSheet: () => void
+  enableEditing: () => void
+  formMethods: UseFormMethods<IBatchRoundDataEntryFormState>
+  index: number
+  initiateSheetRename: () => void
+  isEditing: boolean
+  isRenamingSheet: boolean
+  isTotalsSheet?: boolean
+  savedResults: { [choiceId: string]: number }
+  saveResults: (e: BaseSyntheticEvent) => void
+  sheets: Partial<ArrayField<IBatchResultTallySheet, 'id'>>[]
+}
+
+const BatchResultTallySheet: React.FC<IBatchResultTallySheetProps> = ({
+  areResultsFinalized,
+  cancelSheetRename,
+  confirmSheetRename,
+  contest,
+  deleteSheet,
+  enableEditing,
+  formMethods,
+  index,
+  initiateSheetRename,
+  isEditing,
+  isRenamingSheet,
+  isTotalsSheet,
+  savedResults,
+  saveResults,
+  sheets,
+}) => {
+  const { errors, formState, getValues, register, trigger, watch } = formMethods
+
+  return (
+    <>
+      <BatchResultTallySheetTable>
+        <thead>
+          <tr>
+            <th>Choice</th>
+            <th>Votes</th>
+          </tr>
+        </thead>
+        <tbody>
+          {contest.choices.map(choice => (
```

Comment:
What happens when there are a large number of candidates? I've started to establish a pattern where the step panel is fixed height and scrolling happens internally, so I'm wondering how this component scrolls (if at all)

Reply:
I once again have the most rudimentary scrolling in place, both for long lists of choices and long lists of batches. But know that there's room for improvement, e.g. via sticky elements, like a sticky search bar, sticky tab bar, and maybe even sticky actions. Taking note of this in https://github.com/votingworks/arlo/issues/1655

_Long list of batches_

https://user-images.githubusercontent.com/12616928/198332310-e2289af3-f5ee-409d-b91f-06c7028ffe1e.mov

_Long list of choices_

https://user-images.githubusercontent.com/12616928/198332305-82b5b09b-0e75-4ca6-9658-1c84242d4b6a.mov